src/interactive-window/interactiveWindow.ts

```diff
@@ -222,6 +229,18 @@ export class InteractiveWindow implements IInteractiveWindowLoadable {
         this.start();
     }
 
+    /**
+     * Inform the controller that a cell is being added and it should wait before adding any others to the execution queue.
+     * @param cellAddedPromise - Promise that resolves when the cell execution has been queued
+     */
+    private setPendingCellAdd(cellAddedPromise: Promise<void>) {
```

Comment:
I think we should be passing in the Notebook Document as an argument, so that the promise is scoped to a particular notebook, not all IW that use this controller.

Reply:
added the parameter to the controller's method